runtime/gc_modron_standard/StandardAccessBarrier.cpp

```diff
@@ -1035,3 +1036,22 @@ MM_StandardAccessBarrier::checkClassLive(J9JavaVM *javaVM, J9Class *classPtr)
 }
 #endif /* defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING) */
 
+void
+MM_StandardAccessBarrier::preMountContinuation(J9VMThread *vmThread, j9object_t contObject)
+{
+	if (_extensions->isConcurrentScavengerInProgress()) {
+		/* concurrent scavenger in progress */
+		MM_EnvironmentStandard *env = (MM_EnvironmentStandard *) MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread);
+		MM_ScavengeScanReason reason = SCAN_REASON_SCAVENGE;
+		_scavenger->getDelegate()->doContinuationObject(env, contObject, reason);
```

Comment:
Do you know when thread is mounted first time if J9VMJDKINTERNALVMCONTINUATION_STARTED is returning true or false? Effectively no scan work should be done on first mount (obviously there is not stack yet), and we should realize that as early as possible. At least not to try to iterate stack, but ideally as early as in Interpreter in enterContinuation.
Nothing to do for this round of changes, though.

Reply:
before first time enterContinuation(), J9VMJDKINTERNALVMCONTINUATION_STARTED return false, enterContinuation() will set it true.

Reply:
> before first time enterContinuation(), J9VMJDKINTERNALVMCONTINUATION_STARTED return false, enterContinuation() will set it true.

OK, thanks, that's what I expected. We will still waste time to enter the callback and barrier code, but at least we will not try to setup and enter stack scanning.